### PR TITLE
[SD-3878] set the text component of the dateline if a city is set

### PR DESCRIPTION
**Pull request by marwoodandrew**

### CODE REVIEW

File: superdesk/io/feed_parsers/__init__.py [hunk 1]
```diff
@@ -63,13 +63,17 @@ def set_dateline(self, item, city=None, text=None):
         :type text: str
         """
 
-        item['dateline'] = {}
+        if 'dateline' not in item:
+            item['dateline'] = {}
```
Comment (plamut) [inline superdesk/io/feed_parsers/__init__.py]: For cases like this, the `setdefault()` method comes handy, avoiding the need for the explicit check :wink:
```py
item.setdefault('dateline', {})
```
Cheers!

File: superdesk/io/feed_parsers/__init__.py [hunk 2]
```diff
@@ -63,13 +63,17 @@ def set_dateline(self, item, city=None, text=None):
...
 
         if city:
             cities = find_cities()
             located = [c for c in cities if c['city'] == city]
             item['dateline']['located'] = located[0] if len(located) > 0 else {'city_code': city, 'city': city,
                                                                                'tz': 'UTC', 'dateline': 'city'}
+            if 'text' not in item['dateline']:
```
Comment (mugurrus) [inline superdesk/io/feed_parsers/__init__.py]: Same here, use setdefault

File: superdesk/io/feed_parsers/__init__.py (hunk 1 above)
Comment (mdhaman) [inline superdesk/io/feed_parsers/__init__.py]: +1 welcome back @plamut

Comment (plamut) [inline superdesk/io/feed_parsers/__init__.py]: Well, I'm not actually back, I just dropped by to say hi and wish you guys a (late) happy new year. :wink:

File: superdesk/io/feed_parsers/__init__.py [hunk 3]
```diff
@@ -63,13 +63,15 @@ def set_dateline(self, item, city=None, text=None):
...
 
-        item['dateline'] = {}
+        item.setdefault('dateline', {})
 
         if city:
             cities = find_cities()
             located = [c for c in cities if c['city'] == city]
             item['dateline']['located'] = located[0] if len(located) > 0 else {'city_code': city, 'city': city,
                                                                                'tz': 'UTC', 'dateline': 'city'}
+            item['dateline'].setdefault('text', city)
```
Comment (sivakuna-aap) [inline superdesk/io/feed_parsers/__init__.py]: The `dateline`'s `text` must always have the full format of the dateline i.e. City, Date and Source not just the city.

Comment (marwoodandrew) [inline superdesk/io/feed_parsers/__init__.py]: Problem is what date component and source to use?

I will remove the attempt to set the dateline altogether. It seems to be always Bangalore any way!

**Comment by petrjasek** [comment]
:+1:

**Comment by mugurrus** [comment]
:+1:

**Pull request merged**